tests/WE2E/run_WE2E_tests.sh

```diff
@@ -65,6 +65,63 @@ WE2Edir="$TESTSdir/WE2E"
 #
 #-----------------------------------------------------------------------
 #
+# Run python checks
+#
+#-----------------------------------------------------------------------
+#
+
+# This line will return two numbers: the python major and minor versions
+pyversion=($(/usr/bin/env python3 -c 'import platform; major, minor, patch = platform.python_version_tuple(); print(major); print(minor)'))
+
+#Now, set an error check variable so that we can print all python errors rather than just the first
+pyerrors=0
+
+# Check that the call to python3 returned no errors, then check if the 
+# python3 minor version is 6 or higher
+if [[ -z "$pyversion" ]];then
+  print_info_msg "\
+
+  Error: python3 not found"
+  pyerrors=$((pyerrors+1))
+else
+  if [[ ${#pyversion[@]} -lt 2 ]]; then
+    print_info_msg "\
+
+  Error retrieving python3 version"
+    pyerrors=$((pyerrors+1))
+  elif [[ ${pyversion[1]} -lt 6 ]]; then
+    print_info_msg "\
+
+  Error: python version must be 3.6 or higher
+  python version: ${pyversion[*]}"
+    pyerrors=$((pyerrors+1))
+  fi
+fi
+
+#Next, check for the non-standard python packages: jinja2, yaml, and f90nml
+pkgs=(jinja2 yaml f90nml)
+for pkg in ${pkgs[@]}  ; do
+  if ! /usr/bin/env python3 -c "import ${pkg}" &> /dev/null; then
+  print_info_msg "\
+
+  Error: python module ${pkg} not available"
+  pyerrors=$((pyerrors+1))
+  fi
+done
+
+#Finally, check if the number of errors is >0, and if so exit with helpful message
+if [ $pyerrors -gt 0 ];then
+  print_err_msg_exit "\
+  Errors found: check your python environment
+
+  Instructions for setting up python environments can be found on the web:
+  https://github.com/ufs-community/ufs-srweather-app/wiki/Getting-Started
+
+"
+fi
+#
+#-----------------------------------------------------------------------
+#
```

Comment:
@danielabdi-noaa This script calls get_WE2Etest_names_subdirs_descs.sh which calls $USHdir/calculate_cost.py, which at the very least needs pyyaml (and maybe more, I haven't checked further than that). So we need python checks at the beginning of this script to check for a graceful exit.

Hopefully soon this will be python-based end-to-end so these checks can be neater.